core/lib/instrument_api.h

```diff
@@ -2719,12 +2719,15 @@ void
 dr_mutex_lock(void *mutex);
 
 DR_API
-/** Unlocks \p mutex.  Asserts that mutex is currently locked. */
+/**
+ * Unlocks \p mutex.  Asserts that mutex is currently locked by the
```

Comment:
style: For multi-line doxygen `/**` is on its own line.  I went and ahead and just fixed it since it lets me commit tweaks.

Reply:
Nit: Maybe also add a line of documentation for dr_mutex_lock, stating that the same thread must be used to unlock.(optional suggestion)

Reply:
I added a note to dr_mutex_create which already talks about other restrictions.